ocm_python_wrapper/cluster.py

```diff
@@ -256,33 +257,96 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
-        _info = self.addon_info()
-        _parameters = _info.get("parameters")
-        if not _parameters and parameters:
-            raise ValueError(f"{self.addon_name} does not take any parameters")
+    def filter_required_addon_parameters(self, addon_parameters):
+        """Filters out addon parameter by specified cluster-related conditions.
 
+        Args:
+            addon_parameters (dict) : Addons parameters from api
+
+        Returns: List of dicts of parameters that filtered by matching clusters current configuration and are
+        required for installation.
+        """
+        filtered_parameters = []
         required_parameters = [
-            param["id"] for param in _parameters["items"] if param["required"] is True
+            {
+                "id": param["id"],
+                "default": param.get("default_value"),
+                "conditions": param.get("conditions"),
+            }
```

Comment:
There's no need to use a list of dicts; create a dict where the keys are the names of the params (i.e `id`) and inside have `default_value` and `conditions` keys

Reply:
Done